src/borg/remote.py

```diff
@@ -356,36 +479,38 @@ def pop_preload_msgid(chunkid):
                 del self.chunkid_to_msgids[chunkid]
             return msgid
 
-        def handle_error(error, res):
-            error = error.decode('utf-8')
-            if error == 'DoesNotExist':
+        def handle_error(error, args):
+            if error == b'DoesNotExist':
```

Comment:
The problem here is that error sometimes isn‘t even a string, iirc. The error propagation code is somewhat strange in current borg. Maybe this needs a good look again and not just incremental patching.